scipy/stats/_distn_infrastructure.py

```diff
@@ -3823,7 +3821,7 @@ def _iter_chunked(x0, x1, chunksize=4, inc=1):
         yield supp
 
 
-class rv_sample(rv_discrete):
+class _rv_sample(rv_discrete):
```

Comment:
I hope this won't bother anyone, but it is a backward incompatibility that could be an issue if others looked for `rv_sample` as we did above. We would need to email the mailing list about it and list it in the release notes as a backward incompatibility. Alternatively, we could leave this as `rv_sample` and use a new name for the new class.

Reply:
I'm not sure what is the backcompat issue? `rv_discrete(values=...)` keeps working unchanged, `isinstance(obj, rv_discrete)` keeps working. The specific class name did change, yes, so `obj.__class__.__name__` is different; my understanding is that `isinstance` checks are vastly preferred to explicit `__class__` checks, so I don't think we need to worry about these. What would be other backcompat issues?

That said, if you feel strongly about this, I'm ready to rename it back and use for the public class whatever name you come up with.

Reply:
I think I wasn't thinking straight. `rv_sample` wasn't public before, so there cannot have been an `isinstance` BC issue, and I agree that we don't need to be BC w.r.t. `obj.__class__.__name__`.

Nonetheless, it brings up the idea of choosing a more appropriate name for the class. Is "rv_sample" appropriate? That suggests to me that it is defined by a "sample" as `rv_histogram` is defined by a histogram, but I don't think that would be an accurate interpretation.  TBH `rv_discrete_histogram` would be apt, in a sense, but it may be confusing because the output of `np.histogram` cannot be used directly to instantiate it. `rv_count` or `rv_frequency` might work, especially if we were to accept non-normalized `pk` and normalize it (e.g. `pk /= np.sum(pk)`).